executor/runtime/docker/docker_image.go

```diff
@@ -0,0 +1,70 @@
+package docker
+
+import (
+	"strings"
+
+	"github.com/docker/distribution/reference"
+	"github.com/docker/docker/api/types"
+)
+
+// cleanContainerVersion tries to provide the most human-friendly version string we can get for
+// a particular image, given a version inspect.
+// It goes from most human-friendly to least (sha)
+func cleanContainerVersion(i *types.ImageInspect) string {
+	tag := getImageTagFromInspect(i)
+	if tag != "" {
+		return tag
+	}
+	buildLabel := getBuildLabelFromInspect(i)
+	if buildLabel != "" {
+		return buildLabel
+	}
+	return getImageShaFromInspect(i)
+}
+
+func getImageTagFromInspect(i *types.ImageInspect) string {
+	if len(i.RepoTags) < 1 {
+		return ""
+	}
+	return i.RepoTags[0]
+}
+
+func getBuildLabelFromInspect(i *types.ImageInspect) string {
+	if i.Config == nil {
+		return ""
+	}
+	if i.Config.Labels == nil {
+		return ""
+	}
+	nameLabel, ok := i.Config.Labels["image-name"]
+	if ok {
+		versionLabel, ok := i.Config.Labels["image-version"]
+		if ok {
+			return "image:" + fullImageNameToShortName(nameLabel) + " build:" + versionLabel
+		}
+	}
+	return ""
+}
+
+func getImageShaFromInspect(i *types.ImageInspect) string {
+	if len(i.RepoDigests) < 1 {
+		return ""
+	}
+	name := fullImageNameToShortName(i.RepoDigests[0])
+	splitDigest := strings.Split(i.RepoDigests[0], "@")
+	if len(splitDigest) == 0 {
```

Comment:
`strings.Split` won't return a zero-length slice:

> If s does not contain sep and sep is not empty, Split returns a slice of length 1 whose only element is s.

Reply:
I think you want to check `if len(splitDigest) < 2` instead to make the use of `splitDigest[1]` below safe.